enhancements/proposals/disk_setup.md

```diff
@@ -0,0 +1,186 @@
+
+```
+
+type ScyllaOperatorConfigDiskBenchmark struct {
+    Enabled bool `json:"enabled"`
+    Force bool `json:"force"`
+}
+
+type ScyllaOperatorConfigDiskSetup struct {
+    InstallDiskProvisioner bool `json:"installDiskProvisioner"`
+    LocalDisksHostPath string `json:"localDisksHostPath"`
+    SetupLocalDisks bool `json:"setupLocalDisks"`
+}
+
+type ScyllaOperatorConfigSpec struct {
+    DiskSetup ScyllaOperatorConfigDiskSetup `json:"diskSetup"`
+    DiskBenchmark ScyllaOperatorConfigDiskBenchmark `json:"diskBenchmark"`
+}
+
+```
+
+`DiskSetup.InstallDiskProvisioner` - `true` by default - controls if Operator should install and use Disk Provisioner. 
+Users using their own solution for exposing PVs should disable it.
+
+`DiskSetup.SetupLocalDisks` - `true` by default - controls if NodeTuner DaemonSet should detect and create RAID0 array and format partitions to XFS.
+Users using their own solution for formatting drives should disable it.
+
+`DiskSetup.LocalDisksHostPath` - `/mnt/raid-disks` by default - path on host where raid drive will be created. 
+Disk Provisioner is going to use this path to watch for disks to expose.
+Users using their own Disk Provisioner should sync this path with their installation.
+
+`DiskBenchmark.Enabled` - `true` by default - controls if NodeTuner DaemonSet should perform disk benchmark on PVs attached to Scylla Pods.
+If this is disabled, Scylla Pods doesn't wait for benchmark results, but they trigger it every startup.
+
+`DiskBenchmark.Force` - `false` by default - control if NodeTuner DaemonSet should always run the benchmark if it doesn't exists, and shouldn't 
+use precomputed values for known instances. Underneath it's going to run `iotune` binary on it's own, instead running `scylla_io_setup` script.
+
+### RBAC
+
+ClusterRole bound to ServiceAccount used by NodeTuner DaemonSet is going to be extended with read permissions to PersistentVolumes and patch permission to Nodes.
+
+Operator will provision a new ServiceAccount for Disk Provisioner DaemonSet with `system:persistent-volume-provisioner` ClusterRole.
+
+### Flow
+
+Scylla Operator is going to reconcile Disk Provisioner and NodeTuner DaemonSet based on values in available ScyllaOperatorConfig.
+It's going to create a NodeTuner DaemonSet even when no Scylla Pods are running.
+
+If `ScyllaOperatorConfig.DiskSetup.SetupLocalDisks` is true NodeTuner DaemonSet is going to discover disks attached 
+to the instance and will setup RAID0, format to XFS and will mount it in the location specified in `ScyllaOperatorConfig.DiskSetup.LocalDisksHostPath`.
+
+When new disk appears in specified location, Disk Provisioner will expose them as PersistentVolumes.
+
+Node Tuner DaemonSet is going to have a special finalizer in PodTemplateSpec. Operator will add a special label to each Node
+running Node Tuner Pod.
+When Node Tuner Pod is evicted, Operator will first remove label from the Node, and then clear the finalizer.
```

Comment:
if a DS pod is evicted a new one is created in its place, so eviction isn't the event to trigger on